dcos/dcos-ui
# DCOS-48262 Manual update/reset UX

Adding the `ui-update` plugin that will be used to display the currently
installed ui version, as well as interaction with the new
dcos-ui-update-service. A user will be able to rollback updates as well
as install new versions of the ui if a newer version exists in cosmos.

Closes DCOS-48262

## Testing

Testing this PR requires an OSS cluster that is running the latest `dcos-ui-update-service`. You can deploy a cluster from `DCOS PR#4345` or if that has merged into `dcos` `master` a cluster from master.

1. Add the following package repository to your cluster:
`https://universe-converter.mesosphere.com/transform?url=https://s3-us-west-2.amazonaws.com/rnorris-test-files/dcos-ui-update-testing.json` with priority 0.
This stub universe points to the `dcos-ui` nightly build with a package version of `2.500.0` to allow testing a "new" package.
2. Navigate to the UI Settings page `/#/settings/ui-settings`
3. Wait for versions to load
4. Confirm you see "Available Version 2.500.0"
5. Click "Start Update"
6. Wait for update to complete
7. Confirm Available update row is not shown when update completes
8 Confirm "Refresh to load (X.X.X)" <- this build is the actual build # not the fake 2.500.0 if the package & build versions match these numbers would match
9. Refresh the Page
10. Confirm new installed version is displayed
11. Confirm "Available Version" is not displayed

After testing an update you can test a rollback
1.  Navigate to the UI Settings page `/#/settings/ui-settings`
2. Wait for versions to load
3. Note the Installed Version #
4. Click Rollback
5. Wait for rollback to complete
6. Confirm "Refresh to load (X.X.X)" is shown when rollback is compelte
7. Refresh the page
8. Confirm Installed Version is different
9. Confirm "Available Version" is displayed again

## Trade-offs



## Dependencies


✅ [DCOS 4345](https://github.com/dcos/dcos/pull/4345) ~needs to land before this can merged. You can deploy a cluster against that PR to test though.~

## Screenshots

![image](https://user-images.githubusercontent.com/1972968/53127371-e4322900-3527-11e9-9b3c-a6d66788f7a0.png)
![image](https://user-images.githubusercontent.com/1972968/53127281-b64ce480-3527-11e9-86e9-01e6796b8dff.png)
![image](https://user-images.githubusercontent.com/1972968/53127441-0fb51380-3528-11e9-8ac0-4b881b62d8f2.png)
![image](https://user-images.githubusercontent.com/1972968/53127475-278c9780-3528-11e9-9afb-97a2a6103421.png)
![image](https://user-images.githubusercontent.com/1972968/53127532-4723c000-3528-11e9-81d4-a537b230f2d2.png)
